### ISSUE TITLE
allow creation on remote (X11) displays

### ISSUE THREAD

**Issue by umlaeute** [glfw/glfw#2547]
with X11 it is possible to open a window on a remote display.

i have a some users who are using this, allowing them to display the control interface on one computer, and the openGL output window on another computer - all from within one application. and they might even decide to switch the target display during runtime (obviously destroying the window first, then re-creating it on another `DISPLAY`)

now that we have switched our default windowing backend from GLX to GLFW, this is no longer possible.

i did a quick check to see whether I could trick glfw by setting the `DISPLAY` variable before calling `glfwCreateWindow()`, but alas! this does not work (as glfw requests the display only once, during `glfwInit()`, which can only be called once).

would it be possible to add a (possibly X11 specific) Window creation hint (e.g. `glfwWindowHintString(GLFW_DISPLAY, ":0")`) to allow setting a target display?

i'm aware that currently glfw maintains a global display for all its windows....

**Comment by dougbinks** [comment]
GLFW has a number of platform specific hints already, so a PR for this would likely be acceptable.

**Comment by umlaeute** [comment]
> i'm aware that currently glfw maintains a global display for all its windows....

given the above, should a PR try to keep the current behavior (so selecting a display can only be done before the first window is created) or maintain a display for each window (which might get tricky if the different displays have different capabilities).

i tend towards the former, but then I think the proper function would be `glfwInitHintString()` (which does not exist yet!).

(note that i haven't checked the code recently, so i might have gotten things wrong)

**Comment by dougbinks** [comment]
Out of interest can you rework the application to call init after changing the `DISPLAY` environment variable?

A change to add per window displays would be pretty large, since the GLFW init uses the display variable to initialize a large number of variables used by all the systems in GLFW. So if a change to add this as an init hint would work for your use case then I think this would be the best way to go.

**Comment by umlaeute** [comment]
> Out of interest can you rework the application to call init after changing the `DISPLAY` environment variable?

probably.

so do I understand correctly, that a PR should add a new `glfwInitHintString()` function (because the existing `glfwInitHint()` cannot take strings and is therefore not sufficient)

**Comment by dougbinks** [comment]
Yes, a new function `glfwInitHintString(int hint, const char* value)` similar to `glfwWindowHintString(int hint, const char* value)` looks like the way to do this.
